### PR TITLE
Recipe google analytics

### PR DESCRIPTION
**Pull request by jeffpascal**
## Description

Recipe for issue #20384 

### Documentation

Included steps: 

1: Verify the domain in search.google.com
2: Linking the domain to Google analytics admin
3: Getting the analytics tracking id
4: Using the id in the plugin

Related to #20384

### CODE REVIEW

**Comment by jeffpascal** [comment]
Hello ! anyone know why my checks are failing?

**Comment by laurieontech** [comment]
I should be able to fix the tests, no worries there. Thanks for the PR and we'll review it shortly!

File: docs/docs/recipes/deploying-your-site.md [hunk 1]
```diff
@@ -130,3 +130,59 @@ Use [`wrangler`](https://developers.cloudflare.com/workers/tooling/wrangler/) to
 ### Additional resources
 
 - [Hosting on Cloudflare](/docs/deploying-to-cloudflare-workers)
+
+## Setting up google analytics
```
Comment (laurieontech) [inline docs/docs/recipes/deploying-your-site.md:134]:
```suggestion
## Setting up Google Analytics
```

File: docs/docs/recipes/deploying-your-site.md [hunk 2]
```diff
@@ -130,3 +130,59 @@ Use [`wrangler`](https://developers.cloudflare.com/workers/tooling/wrangler/) to
...
+
+### Using gatsby-plugin-google-analytics to track the activity of your Gatsby Site
```
Comment (laurieontech) [inline docs/docs/recipes/deploying-your-site.md:136]:
```suggestion
### Using gatsby-plugin-google-analytics
```

File: docs/docs/recipes/deploying-your-site.md [hunk 3]
```diff
@@ -130,3 +130,59 @@ Use [`wrangler`](https://developers.cloudflare.com/workers/tooling/wrangler/) to
...
+
+gatsby-plugin-google-analytics is a very popular plugin that provides insights in the access patters of your website
```
Comment (laurieontech) [inline docs/docs/recipes/deploying-your-site.md:138]:
```suggestion
`gatsby-plugin-google-analytics` is a popular plugin that provides insights into how users access your website. You can use it to track site activity.
```

File: docs/docs/recipes/deploying-your-site.md [hunk 4]
```diff
@@ -130,3 +130,59 @@ Use [`wrangler`](https://developers.cloudflare.com/workers/tooling/wrangler/) to
...
+
+### Prerequisites
+
+- A [Gatsby site](/docs/quick-start) with `gatsby-config.js` and an `index.js` page
```
Comment (laurieontech) [inline docs/docs/recipes/deploying-your-site.md:142]:
```suggestion
- A [Gatsby site](/docs/quick-start) with a `gatsby-config.js` file and an `index.js` page
```

File: docs/docs/recipes/deploying-your-site.md [hunk 5]
```diff
@@ -130,3 +130,59 @@ Use [`wrangler`](https://developers.cloudflare.com/workers/tooling/wrangler/) to
...
+- The [Gatsby CLI](/docs/gatsby-cli) installed
+- A Domain ex: [AWS](https://aws.amazon.com/getting-started/tutorials/get-a-domain/)
```
Comment (laurieontech) [inline docs/docs/recipes/deploying-your-site.md:144]:
```suggestion
- A Domain from your provider of choice,  e.g. [AWS](https://aws.amazon.com/getting-started/tutorials/get-a-domain/)
```

File: docs/docs/recipes/deploying-your-site.md [hunk 6]
```diff
@@ -130,3 +130,59 @@ Use [`wrangler`](https://developers.cloudflare.com/workers/tooling/wrangler/) to
...
+
+#### Verify the domain in search.google.com
```
Comment (laurieontech) [inline docs/docs/recipes/deploying-your-site.md:146]:
```suggestion
### Verify the domain in search.google.com
```

File: docs/docs/recipes/deploying-your-site.md [hunk 7]
```diff
@@ -130,3 +130,59 @@ Use [`wrangler`](https://developers.cloudflare.com/workers/tooling/wrangler/) to
...
+
+1. [Verify the domain](https://search.google.com/search-console/not-verified) by clicking on **Search Property** > **Add Property**. Type in your domain, and press Continue.
```
Comment (laurieontech) [inline docs/docs/recipes/deploying-your-site.md:148]:
```suggestion
1. Navigate to the [Google search console]((https://search.google.com/search-console/not-verified)) to verify the domain by clicking on **Search Property** > **Add Property**. Type in your domain and press Continue.
```

File: docs/docs/recipes/deploying-your-site.md [hunk 8]
```diff
@@ -130,3 +130,59 @@ Use [`wrangler`](https://developers.cloudflare.com/workers/tooling/wrangler/) to
...
+2. Create a **TXT** record in the DNS configuration for example [on AWS](https://www.entrustdatacard.com/knowledgebase/how-to-create-a-txt-record-on-amazon-aws-route-53-for-entrust-email-validation-method)
```
Comment (laurieontech) [inline docs/docs/recipes/deploying-your-site.md:149]:
```suggestion
2. Add a **TXT** record to your DNS configuration. Follow the directions for your provider, or refer to [Google docs](https://support.google.com/a/answer/183895?hl=en).
```

File: docs/docs/recipes/deploying-your-site.md [hunk 9]
```diff
@@ -130,3 +130,59 @@ Use [`wrangler`](https://developers.cloudflare.com/workers/tooling/wrangler/) to
...
+
+#### Linking the domain to Google analytics admin
```
Comment (laurieontech) [inline docs/docs/recipes/deploying-your-site.md:151]:
```suggestion
### Linking the domain to Google Analytics admin
```

File: docs/docs/recipes/deploying-your-site.md [hunk 10]
```diff
@@ -130,3 +130,59 @@ Use [`wrangler`](https://developers.cloudflare.com/workers/tooling/wrangler/) to
...
+
+1. Log in to [Analytics](https://analytics.google.com/analytics/)
```
Comment (laurieontech) [inline docs/docs/recipes/deploying-your-site.md:153]:
```suggestion
1. Log into [Google Analytics](https://analytics.google.com/analytics/).
```

File: docs/docs/recipes/deploying-your-site.md [hunk 11]
```diff
@@ -130,3 +130,59 @@ Use [`wrangler`](https://developers.cloudflare.com/workers/tooling/wrangler/) to
...
+2. Click Admin.
+3. Select **Create Property** in Property column
```
Comment (laurieontech) [inline docs/docs/recipes/deploying-your-site.md:155]:
```suggestion
3. Select **Create Property** in the Property column.
```

File: docs/docs/recipes/deploying-your-site.md [hunk 12]
```diff
@@ -130,3 +130,59 @@ Use [`wrangler`](https://developers.cloudflare.com/workers/tooling/wrangler/) to
...
+4. Choose **Web**
+5. Fill in the details and click Create.
+
+#### Getting the analytics tracking id
```
Comment (laurieontech) [inline docs/docs/recipes/deploying-your-site.md:159]:
```suggestion
### Getting your Google Analytics `Tracking ID`
```

File: docs/docs/recipes/deploying-your-site.md [hunk 13]
```diff
@@ -130,3 +130,59 @@ Use [`wrangler`](https://developers.cloudflare.com/workers/tooling/wrangler/) to
...
+
+1. Sign in to your Analytics account.
```
Comment (laurieontech) [inline docs/docs/recipes/deploying-your-site.md:161]:
```suggestion
1. Sign in to your Google Analytics account.
```

File: docs/docs/recipes/deploying-your-site.md [hunk 14]
```diff
@@ -130,3 +130,59 @@ Use [`wrangler`](https://developers.cloudflare.com/workers/tooling/wrangler/) to
...
+
+1. [Verify the domain](https://search.google.com/search-console/not-verified) by clicking on **Search Property** > **Add Property**. Type in your domain, and press Continue.
+2. Create a **TXT** record in the DNS configuration for example [on AWS](https://www.entrustdatacard.com/knowledgebase/how-to-create-a-txt-record-on-amazon-aws-route-53-for-entrust-email-validation-method)
+
+#### Linking the domain to Google analytics admin
+
+1. Log in to [Analytics](https://analytics.google.com/analytics/)
+2. Click Admin.
+3. Select **Create Property** in Property column
+4. Choose **Web**
```
Comment (laurieontech) [inline docs/docs/recipes/deploying-your-site.md:156]:
```suggestion
4. Choose **Web**.
```

File: docs/docs/recipes/deploying-your-site.md [hunk 15]
```diff
@@ -130,3 +130,59 @@ Use [`wrangler`](https://developers.cloudflare.com/workers/tooling/wrangler/) to
...
+
+#### Getting the analytics tracking id
+
+1. Sign in to your Analytics account.
+2. Click Admin.
+3. Select an account from the menu in the ACCOUNT column.
+4. Select a property from the menu in the PROPERTY column.
+5. Under PROPERTY, click Tracking Info > Tracking Code. Your Tracking ID is displayed at the top of the page.
+
+#### Using the id in the plugin
```
Comment (laurieontech) [inline docs/docs/recipes/deploying-your-site.md:167]:
```suggestion
### Using the ID in the plugin
```

File: docs/docs/recipes/deploying-your-site.md [hunk 16]
```diff
@@ -130,3 +130,59 @@ Use [`wrangler`](https://developers.cloudflare.com/workers/tooling/wrangler/) to
...
+
+1. `npm install --save gatsby-plugin-google-analytics`
```
Comment (laurieontech) [inline docs/docs/recipes/deploying-your-site.md:169]:
```suggestion
1. Run `npm install gatsby-plugin-google-analytics` in your terminal.
```

File: docs/docs/recipes/deploying-your-site.md [hunk 17]
```diff
@@ -130,3 +130,59 @@ Use [`wrangler`](https://developers.cloudflare.com/workers/tooling/wrangler/) to
...
+2. in gatsby-config.js add
```
Comment (laurieontech) [inline docs/docs/recipes/deploying-your-site.md:170]:
```suggestion
2. Add the following to your `gatsby-config.js` file.
```

File: docs/docs/recipes/deploying-your-site.md [hunk 18]
```diff
@@ -130,3 +130,59 @@ Use [`wrangler`](https://developers.cloudflare.com/workers/tooling/wrangler/) to
...
+
+```js
```
Comment (laurieontech) [inline docs/docs/recipes/deploying-your-site.md:172]:
```suggestion
```javascript:title="gatsby-config.js"
```

**Comment by jeffpascal** [comment]
> Thanks so much for your work on this! I left some suggestions below.
> 
> One other question is when things are bolded vs. capitalized or otherwise for actions the user should take. It would be good to be consistent.

Thank you for the feedback/corrections and for reviewing my work

**Comment by laurieontech** [comment]
Awesome! I made some minor changes to align with formatting in other recipes but this looks good to go. Thanks so much for contributing to Gatsby!

**Pull request merged**